workflow/Snakefile

```diff
@@ -0,0 +1,229 @@
+
+
+RESULTS_DIR = config['predictions_results_dir']
+
+def get_accessibility_file(wildcards):
+	default_accessibility_feature = biosamples_config.loc[wildcards.biosample, 'default_accessibility_feature']
+	return biosamples_config.loc[wildcards.biosample, default_accessibility_feature]
+
+## overall rule to run pipeline
+rule all:
+	input:
+		narrowPeaks_sorted = expand(os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted"), biosample=all_biosamples),
+		candidateRegions = expand(os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted.candidateRegions.bed"),
+	    biosample=all_biosamples),
+		enhancerLists = expand(os.path.join(RESULTS_DIR, "{biosample}", "Neighborhoods", "EnhancerList.txt"),  biosample=all_biosamples),
+		allPutative = expand(os.path.join(RESULTS_DIR, "{biosample}", "Predictions", "EnhancerPredictionsAllPutative.txt.gz"), biosample=all_biosamples),
+		enhPredictions = expand(os.path.join(RESULTS_DIR, "{biosample}", "Predictions", "EnhancerPredictions.csv"), biosample=all_biosamples),
+		allPred = os.path.join(RESULTS_DIR,"AllPredictions.txt.gz"),
+		altTSS = TSS_files,
+		metricsDirectory = expand(os.path.join(RESULTS_DIR, "{biosample}", "Metrics", "QCSummary.p"), biosample=all_biosamples)
+
+	
+## call macs2 -- if multiple accessibility inputs for one biosample, will aggregate into one output
+rule call_macs_peaks: 
+	input:
+		accessibility = get_accessibility_file
+	params:
+		pval = config['params_macs']['pval'],
+		out_dir = config["predictions_results_dir"]
+	conda:
+		"envs/abcenv.yml"
+	output: 
+		narrowPeak = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak")
+	shell: 
+		""" 
+		macs2 callpeak -f AUTO -g hs -p {params.pval} -n macs2 --call-summits --outdir {params.out_dir}/{wildcards.biosample}/Peaks -t {input.accessibility}
+		"""
+
+## sort narrowPeaks
+rule sort_narrowpeaks:
+	input:
+		narrowPeak = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak")
+	params:
+		chrom_sizes = config['params_candidate']['chrom_sizes']
+	conda:
+		"envs/abcenv.yml"
+	output:
+		narrowPeakSorted = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted")
+	shell:
+		"""
+		grep -v -E "^\w+_.*" {input.narrowPeak} | bedtools sort -faidx {params.chrom_sizes} -i stdin > {output.narrowPeakSorted}
+		"""
+
+## call candidate regions
+rule call_candidate_regions:
+	input:
+		narrowPeak = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted"),
+	params:
+		accessibility = get_accessibility_file,
+		TSS = lambda wildcards: biosamples_config.loc[wildcards.biosample, 'TSS'],
+		chrom_sizes = config['params_candidate']['chrom_sizes'],
+		regions_blocklist = config['params_candidate']['regions_blocklist'],
+		peakExtendFromSummit = config['params_candidate']['peakExtendFromSummit'],
+		nStrongestPeak = config['params_candidate']['nStrongestPeaks'],
+		threads = 10,
```

Comment:
threads = 10 probably should be a user-controllable parameter

Reply:
lol it doesn't even get passed into the script. removing it